pystac_client/item_search.py

```diff
@@ -591,14 +594,40 @@ def get_items(self) -> Iterator[Item]:
                 if self._max_items and nitems >= self._max_items:
                     return
 
+    def get_items_as_dicts(self) -> Iterator[Dict[str, Any]]:
+        """Iterator that yields :class:`dict` instances for each item matching
+        the given search parameters. Calls
+        :meth:`ItemSearch.get_item_collections()` internally and yields from
+        :attr:`ItemCollection.features <pystac_client.ItemCollection.features>` for
+        each page of results.
+
+        Yields:
+            Item : each Item matching the search criteria
+        """
+        nitems = 0
+        for page in self._stac_io.get_pages(
+            self.url, self.method, self.get_parameters()
+        ):
+            for item in page.get("features", []):
+                yield item
+                nitems += 1
+                if self._max_items and nitems >= self._max_items:
+                    return
+
     @lru_cache(1)
-    def get_all_items_as_dict(self) -> Dict:
-        """Convenience method that gets all items from all pages, up to self._max_items,
-         and returns an array of dictionaries
+    def get_all_items_as_dict(self) -> Dict[str, Any]:
+        """DEPRECATED. Use get_items or get_itemcollections instead.
+            Convenience method that gets all items from all pages, up to
+            self._max_items, and returns an array of dictionaries.
```

Comment:
This isn't introduced in this PR, but I don't love referencing a "private" attribute in the documentation. `_max_items` should either be made public or exposed via a property or method.